lib/uniris/networking/ip_lookup/ip_lookup.ex

```diff
@@ -1,16 +1,16 @@
 defmodule Uniris.Networking.IPLookup do
   @moduledoc false
 
-  alias __MODULE__.Config
-
   # Public
 
-  @spec get_node_ip() :: {:ok, :inet.ip_address()} | {:error, binary}
+  @spec get_node_ip() :: {:ok, :inet.ip_address()} | {:error, :invalid_ip_provider | :not_recognizable_ip | :ip_discovery_error}
   def get_node_ip do
-    with {:ok, ip_provider} <- Config.ip_provider() do
+    with config <- Application.get_env(:uniris, Uniris.Networking),
```

Comment:
One suggestion could be if the provider is not the Static one (defined IP address), try to use Nat(UpnP) otherwise fallback to IPFY maybe.